src/main/java/org/folio/search/model/ResourceEventBody.java

```diff
@@ -0,0 +1,47 @@
+package org.folio.search.model;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+/**
+ *
+ */
+@Data
+@NoArgsConstructor
+@AllArgsConstructor(staticName = "of")
+public class ResourceEventBody {
+
+  /**
+   * Operation type.
+   */
+  private String type;
+
+  /**
+   * Tenant id.
+   */
+  private String tenant;
+
+  /**
+   * Resource name.
+   */
+  private String resourceName;
+
+  /**
+   * Data from update.
+   */
+  @JsonProperty("new")
+  private ObjectNode newData;
```

Comment:
It could looks fruitful, but I don't think, that it would be beneficial to migrate current code to Map, because ResourceEventBody is internal class and won't be used elsewhere. And in addition to that there a plenty of code, which relies on Jackson. So we assume we will use it in this module and can rely on it.

Reply:
Shure, works for me.